examples/documentation_examples/examples.go

```diff
@@ -2097,3 +2097,645 @@ func ChangeStreamExamples(t *testing.T, db *mongo.Database) {
 
 	atomic.StoreInt32(&stop, 1)
 }
+
+// AggregationExamples contains examples of aggregation operations.
+func AggregationExamples(t *testing.T, db *mongo.Database) {
+	ctx := context.Background()
+
+	salesColl := db.Collection("sales")
+	airlinesColl := db.Collection("airlines")
+	airAlliancesColl := db.Collection("air_alliances")
+
+	{
+		// Start Aggregation Example 1
+		docs := []interface{}{
+			bson.D{
+				{"date", bson.D{
+					{"$date", "2018-02-08T09:00:00.000Z"},
+				}},
+				{"items", bson.A{
+					bson.D{
+						{"fruit", "kiwi"},
+						{"quantity", 2},
+						{"price", 0.5},
+					},
+					bson.D{
+						{"fruit", "apple"},
+						{"quantity", 1},
+						{"price", 1.0},
+					},
+				}},
+			},
+			bson.D{
+				{"date", bson.D{
+					{"$date", "2018-01-09T07:12:00.000Z"},
+				}},
+				{"items", bson.A{
+					bson.D{
+						{"fruit", "banana"},
+						{"quantity", 8},
+						{"price", 1.0},
+					},
+					bson.D{
+						{"fruit", "apple"},
+						{"quantity", 1},
+						{"price", 1.0},
+					},
+					bson.D{
+						{"fruit", "papaya"},
+						{"quantity", 1},
+						{"price", 4.0},
+					},
+				}},
+			},
+			bson.D{
+				{"date", bson.D{
+					{"$date", "2018-01-27T09:13:00.000Z"},
+				}},
+				{"items", bson.A{
+					bson.D{
+						{"fruit", "banana"},
+						{"quantity", 1},
+						{"price", 1.0},
+					},
+				}},
+			},
+			bson.D{
+				{"date", bson.D{
+					{"$date", "2018-02-03T07:58:00.000Z"},
+				}},
+				{"items", bson.A{
+					bson.D{
+						{"fruit", "banana"},
+						{"quantity", 1},
+						{"price", 1.0},
+					},
+				}},
+			},
+			bson.D{
+				{"date", bson.D{
+					{"$date", "2018-02-05T06:03:00.000Z"},
+				}},
+				{"items", bson.A{
+					bson.D{
+						{"fruit", "banana"},
+						{"quantity", 1},
+						{"price", 1.0},
+					},
+					bson.D{
+						{"fruit", "mango"},
+						{"quantity", 2},
+						{"price", 2.0},
+					},
+					bson.D{
+						{"fruit", "apple"},
+						{"quantity", 1},
+						{"price", 1.0},
+					},
+				}},
+			},
+			bson.D{
+				{"date", bson.D{
+					{"$date", "2018-01-11T07:15:00.000Z"},
+				}},
+				{"items", bson.A{
+					bson.D{
+						{"fruit", "banana"},
+						{"quantity", 1},
+						{"price", 1.0},
+					},
+					bson.D{
+						{"fruit", "apple"},
+						{"quantity", 1},
+						{"price", 1.0},
+					},
+					bson.D{
+						{"fruit", "papaya"},
+						{"quantity", 3},
+						{"price", 4.0},
+					},
+				}},
+			},
+		}
+
+		result, err := salesColl.InsertMany(ctx, docs)
+
+		// End Aggregation Example 1
+
+		require.NoError(t, err)
+		require.Len(t, result.InsertedIDs, 6)
+	}
+	{
+		// Start Aggregation Example 2
+		pipeline := mongo.Pipeline{
+			{
+				{"$match", bson.D{
+					{"items.fruit", "banana"},
+				}},
+			},
+			{
+				{"$sort", bson.D{
+					{"date", 1},
+				}},
+			},
+		}
+
+		cursor, err := salesColl.Aggregate(ctx, pipeline)
+
+		// End Aggregation Example 2
+
+		require.NoError(t, err)
+		defer cursor.Close(ctx)
+
+		ok := cursor.Next(ctx)
```

Comment:
This applies to every `Aggregate` example:

Do we know how many documents the `Aggregate` is expected to return? If so, can we use `requireCursorLength` to do an exact check rather than only checking that it returns at least 1?

If, for some reason, we don't know the exact count, I'm OK with leaving this as-is, but please remove the blank line between `ok := ...` and `require.True...`. We try to use blank lines to separate logical code blocks, but assigning and checking a value logically belong in the same block.

Reply:
Yep! Replaced the `ok := cursor.Next(ctx); require.True(t, ok)` with `requireCursorLength(t, cursor, n)` where `n` is the number of docs still left in the cursor. And good to know about the new lines!